components/classes/StudentsPrintOptions.tsx

```diff
@@ -74,6 +76,27 @@ const StudentsPrintOptions: FC<{
             </FormItem>
           ))}
         </FormGroup>
+
+        <FormGroup label={t("filterColumns.label")}>
+          {(
+            [
+              mysk.user &&
+                (mysk.user.is_admin || mysk.user.role !== UserRole.student) &&
+                "noElective",
+                "hasAllergies",
+            ].filter((column) => column) as OptionsType["columns"]

```

Comment:
Here’s a clear reason why you should separate `filters` out of `columns`. If you were to put keys from `columns` into this array, TypeScript wouldn’t flag it as an error.